// vue-onsenui/esm/components/VOnsAlertDialog.js
import 'onsenui/esm/elements/ons-alert-dialog';
import { hasOptions, dialogCancel, deriveEvents, deriveDBB, portal, modifier } from '../mixins';

//

var script = {
  name: 'v-ons-alert-dialog',
  mixins: [hasOptions, dialogCancel, deriveEvents, deriveDBB, portal, modifier],

  props: {
    title: {
      type: String
    },
    footer: {
      type: Object,
      validator(value) {
        return Object.keys(value).every(key => value[key] instanceof Function);
      }
    }
  }
};

/* script */
            const __vue_script__ = script;
            
/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('ons-alert-dialog',_vm._g({attrs:{"modifier":_vm.normalizedModifier}},_vm.unrecognizedListeners),[_c('div',{staticClass:"alert-dialog-title"},[_vm._t("title",function(){return [_vm._v(_vm._s(_vm.title))]})],2),_vm._v(" "),_c('div',{staticClass:"alert-dialog-content"},[_vm._t("default")],2),_vm._v(" "),_c('div',{staticClass:"alert-dialog-footer"},[_vm._t("footer",function(){return _vm._l((_vm.footer),function(handler,key){return _c('ons-alert-dialog-button',{key:key,on:{"click":handler}},[_vm._v(_vm._s(key))])})})],2)])};
var __vue_staticRenderFns__ = [];

  /* style */
  const __vue_inject_styles__ = undefined;
  /* scoped */
  const __vue_scope_id__ = undefined;
  /* functional template */
  const __vue_is_functional_template__ = false;
  /* component normalizer */
  function __vue_normalize__(
    template, style, script,
    scope, functional, moduleIdentifier,
    createInjector, createInjectorSSR
  ) {
    const component = (typeof script === 'function' ? script.options : script) || {};

    // For security concerns, we use only base name in production mode.
    component.__file = "VOnsAlertDialog.vue";

    if (!component.render) {
      component.render = template.render;
      component.staticRenderFns = template.staticRenderFns;
      component._compiled = true;

      if (functional) component.functional = true;
    }

    component._scopeId = scope;

    return component
  }
  /* style inject */
  
  /* style inject SSR */
  

  
  var VOnsAlertDialog = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__);

export { VOnsAlertDialog as default };
